src/buildercore/bootstrap.py

```diff
@@ -429,6 +429,32 @@ def download_master_builder_key(stackname):
         operations.get(remote_path=private_key, local_path=fh, use_sudo=True)
     return fh
 
+def download_master_configuration(master_stack):
+    fh = StringIO()
+    with stack_conn(master_stack):
+        operations.get(remote_path='/etc/salt/master.template', local_path=fh, use_sudo=True)
+    return fh
```

Comment:
this is returning a file-like object and not it's string value. is this intended? 
if not, you might want: `return fh.getvalue()` instead

https://docs.python.org/2/library/stringio.html#StringIO.StringIO.getvalue